lib/movements.js

```diff
@@ -92,6 +135,15 @@ class Movements {
     for (const shape of b.shapes) {
       b.height = Math.max(b.height, pos.y + dy + shape[4])
     }
+
+    // check if open door, if so treat it like air so the bot can navigate
+    if (b.name.includes('door') && b.material === 'wood' && doorOpenMetadata.includes(b.metadata)) {
```

Comment:
Seems like this door function should be written elsewhere, like maybe prismarine-block or prismarine-world, so it can be used within mineflayer also. The hard coded data values seem version specific, so adding this logic elsewhere and exposing it as a property on block might be a good idea.

Reply:
The data for the door is already in `block.getProperties()`, you can see the keys available here: https://minecraft.gamepedia.com/Door#Block_data

Also, this code doesn't seem to take into account the direction of the block: if you travel N-S or E-W, the same door could appear open or closed.